### DISCUSSION TITLE
/rename to re-name a player just for yourself

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#7346] [Enhancement]
We have /nick and /realname, but I cant see a way to set a name for someone, just for me, So i can easily keep track of whos who, as some people change their names and I only remember their first username they had!
Bonus points if it can be a staff permissions, so staff can nickname players that only staff can see with a permission.

**Comment by mrfloris** [comment]
<img width="600" alt="Screen-Shot 2022-11-09 at 08 08 37" src="https://user-images.githubusercontent.com/28841349/200762523-67bc0a24-bae5-49ca-b9a8-4b385c65f609.png">

<img width="499" alt="Screen-Shot 2022-11-09 at 08 09 00" src="https://user-images.githubusercontent.com/28841349/200762583-9dd90f11-b1e0-4a7a-aa8e-c13691ebfe13.png">

You can use /cmi nick to change the nickname of a player that's not yourself.

**Comment by TomLewis** [comment]
Missed a key piece of information from the ticket. **just for me**.

Im not sure how to reparse it as my initial ticket is clear! Maybe you skimmed over the request, so like personal nicknames for people not the entire server!

So if I did 
`/nick mrfloris Floris` then you would be shown as Floris to only peolpe with that permission.

**Comment by Zrips** [comment]
Got basic idea here, thing is that having dedicated list of player names just based on receiving end would complicate things quite a bit, as now we would need to have additional column in database and properly update players name for this specific receiver if it should be changed. This can go wild with things like placeholders which generally return players name based on players object and not based on senders, which on itself is a complication as not always you might want to actually modify this based on sender.
In short: To complex for such small gains.

**Comment by TomLewis** [comment]
Add a new placeholder! Then its up to the admin to do the visuals in their chat plugin! ChatControl allows showing specific views based on the player seeing the chat. So for example staff on the server can see specific things, then staff could nickname players and would only be shown to staff via their chat placeholder.
